### ISSUE TITLE
Linter: Support composite keys

### ISSUE THREAD

**Issue by anamargaridarl** [datacontract/cli#138]
Currently, the linter doesn't support all composite keys since they can be created where each column is not unique, but the combination of them is. 

I would like to modify the linter to only check for the unique tag when the primary key is made up of a single column in the table. 

However, I'm facing issues while trying to push my branch into the repository. It seems to be related to a lack of authorization. Could you please help me understand what's happening and let me know if this feature is something you're interested in?

**Comment by jochenchrist** [comment]
> However, I'm facing issues while trying to push my branch into the repository. It seems to be related to a lack of authorization. Could you please help me understand what's happening and let me know if this feature is something you're interested in?

We had an issue in the CI task, when docker tries to log in, this is fixed on the main branch. Try to rebase/merge from main. If not, can you share some more details?

**Comment by jeeger** [comment]
From my understanding of the datacontract spec, the data contract specification does not support specifying unique constraints on multiple fields at all (i.e. `unique` is just a boolean value for each field).
Constraining uniqueness for composite keys is possible via quality checks, but these are not connected to the linting process. Quality checks are run when the `test` subcommand is used.

> I would like to modify the linter to only check for the unique tag when the primary key is made up of a single column in the table.

Ah, this might be a special case. If there are multiple fields with the `primary` value set to true, the only valid interpretation is a composite primary key. This could be a special case in the linter.

However, I am unsure what your linter error message is, could you provide it?

**Comment by anamargaridarl** [comment]
While running the check, I misinterpreted the error message to believe that the linter enforced in all primary keys both the required and unique constraints. In reality, only one constraint is enforced.

`Field 'x' in model 'x' is marked as primary, but not as unique and required.`

However, should the linter allow tables with only one primary key to be created with only required and not uniqueness? 

Regarding the authorization, I think I might be missing repository grants to do it since I already tried to merge with the most recent version.

> ERROR: Permission to datacontract/cli.git denied to anamargaridarl.
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
